File: Haskell/fizzbuzz.hs
-- fizzbuzz.hs

fizzbuzz n = 
  map (\x -> 
    if       x `mod` 15 == 0 then "FizzBuzz"
    else if x `mod`  3 == 0 then "Fizz"
    else if x `mod`  5 == 0 then "Buzz"
    else                          (show x))
  [1..n]
                     
main=print $ fizzbuzz 100

File: JavaScript/fizzbuzz.js
// fizzbuzz.js

for (i =0; i < 100; i++) {
  console.log(
    function(x) {
      if (x % 3 === 0 && x % 5 ===  0) {
        return "FizzBuss";
      } else if (x % 3 === 0) {
        return "Fizz";
      } else if (x % 5 === 0) {
        return "Buzz";
      } else {
        return x;
      }
  }(i));
}


